src/plugins/discover/public/application/components/discover.tsx

```diff
@@ -203,6 +206,14 @@ export function Discover({
     [opts, state]
   );
 
+  const onEditRuntimeField = () => {
+    if (refreshAppState) {
+      refreshAppState();
```

Comment:
If possible we should not introduce new $scope variables in `discover.js`, so by doing it the following way:

```
opts.refetch$.next();
```
this doesn't need to be refactored and removed later